solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/inbound/acknowledge/TransactedJCSMPAcknowledgementCallback.java

```diff
@@ -0,0 +1,98 @@
+package com.solace.spring.cloud.stream.binder.inbound.acknowledge;
+
+import com.solace.spring.cloud.stream.binder.util.ErrorQueueInfrastructure;
+import com.solace.spring.cloud.stream.binder.util.SolaceAcknowledgmentException;
+import com.solacesystems.jcsmp.JCSMPException;
+import com.solacesystems.jcsmp.transaction.RollbackException;
+import com.solacesystems.jcsmp.transaction.TransactedSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.integration.acks.AcknowledgmentCallback;
+
+class TransactedJCSMPAcknowledgementCallback implements AcknowledgmentCallback {
+	private final ThreadLocal<TransactedSession> transactedSessionThreadLocal = new ThreadLocal<>();
+	private final ErrorQueueInfrastructure errorQueueInfrastructure;
+	private boolean acknowledged = false;
+	private static final Logger LOGGER = LoggerFactory.getLogger(TransactedJCSMPAcknowledgementCallback.class);
+
+	TransactedJCSMPAcknowledgementCallback(TransactedSession transactedSession,
+										   ErrorQueueInfrastructure errorQueueInfrastructure) {
+		this.transactedSessionThreadLocal.set(transactedSession);
```

Comment:
setting of  threadlocal in a constructor should be avoided
Thread will likely be not go away, as there is no to clear thread local from a session

Reply:
See previous comment. I'll remove the `ThreadLocal` usage.

Reply:
Replaced usage of `ThreadLocal` to instead use a thread ID cached during object creation.